application/helpers/admin/import_helper.php

```diff
@@ -1891,34 +1891,30 @@ function CSVImportResponses($sFullFilePath, $iSurveyId, $aOptions = array())
     if (!isset($aOptions['bDeleteFistLine'])) {$aOptions['bDeleteFistLine'] = true; } // By default delete first line (vvimport)
     if (!isset($aOptions['sExistingId'])) {$aOptions['sExistingId'] = "ignore"; } // By default exclude existing id
     if (!isset($aOptions['bNotFinalized'])) {$aOptions['bNotFinalized'] = false; } // By default don't change finalized part
-    if (!isset($aOptions['sCharset']) || !$aOptions['sCharset']) {$aOptions['sCharset'] = "utf8"; }
+    if (empty($aOptions['sCharset'])) {
+        $aOptions['sCharset'] = "utf8";
+    }
     if (!isset($aOptions['sSeparator'])) {$aOptions['sSeparator'] = "\t"; }
     if (!isset($aOptions['sQuoted'])) {$aOptions['sQuoted'] = "\""; }
     // Fix some part
-    if (!array_key_exists($aOptions['sCharset'], aEncodingsArray())) {
-        $aOptions['sCharset'] = "utf8";
-    }
 
     // Prepare an array of sentence for result
     $CSVImportResult = array();
-    // Read the file
-    $handle = fopen($sFullFilePath, "r"); // Need to be adapted for Mac ? in options ?
-    if ($handle === false) {
-        safeDie("Can't open file");
+    $tmpVVFile = fileCsvToUtf8($sFullFilePath, $aOptions['sCharset']);
```

Comment:
If this function already reads the file, and this function is only used here (guessing), why don't this funciton just returns an array?

Reply:
Think it can be used elsewhere after : in TXT import/export for example.

Else : function does one think : return an utf8 file :)

Reply:
Else : 2 places : here + TSV import (taken from TSVImportSurvey function)

https://github.com/LimeSurvey/LimeSurvey/pull/2177/files#diff-4033daa1797e130ed555f4b345ed46387aa79c5eb457adf216ea049b648e5976R2222